pkg/operator/rules.go

```diff
@@ -33,39 +33,60 @@ func GenerateRulesConfiguration(promRule monitoringv1.PrometheusRuleSpec, logger
 	if err != nil {
 		return "", errors.Wrap(err, "failed to marshal content")
 	}
-	errs := ValidateRule(promRule)
-	if len(errs) != 0 {
+	admissionErrors := ValidateRule(promRule)
+	if len(admissionErrors) != 0 {
 		const m = "Invalid rule"
 		level.Debug(logger).Log("msg", m, "content", content)
-		for _, err := range errs {
-			level.Info(logger).Log("msg", m, "err", err)
+		for _, admissionError := range admissionErrors {
+			level.Info(logger).Log("msg", m, "field", admissionError.Field, "err", admissionError.Error)
 		}
 		return "", errors.New(m)
 	}
 	return string(content), nil
 }
 
 // ValidateRule takes PrometheusRuleSpec and validates it using the upstream prometheus rule validator.
-func ValidateRule(promRule monitoringv1.PrometheusRuleSpec) []error {
+func ValidateRule(promRule monitoringv1.PrometheusRuleSpec) []AdmissionError {
+	groupTag := GetFieldJSONTag(monitoringv1.PrometheusRuleSpec{}, "Groups")
+	groupMap := make(map[string]int, len(promRule.Groups))
+
 	for i, group := range promRule.Groups {
+		groupMap[group.Name] = i
 		if group.PartialResponseStrategy == "" {
 			continue
 		}
 		// TODO(slashpai): Remove this validation after v0.65 since this is handled at CRD level
 		if _, ok := thanostypes.PartialResponseStrategy_value[strings.ToUpper(group.PartialResponseStrategy)]; !ok {
-			return []error{
-				fmt.Errorf("invalid partial_response_strategy %s value", group.PartialResponseStrategy),
-			}
+			return []AdmissionError{{
+				Field: fmt.Sprintf("%s[%d].%s", groupTag, i, GetFieldJSONTag(monitoringv1.RuleGroup{}, "PartialResponseStrategy")),
+				Error: fmt.Errorf("invalid partial_response_strategy %s value", group.PartialResponseStrategy),
+			}}
 		}
 
 		// reset this as the upstream prometheus rule validator
 		// is not aware of the partial_response_strategy field.
 		promRule.Groups[i].PartialResponseStrategy = ""
 	}
+
 	content, err := yaml.Marshal(promRule)
 	if err != nil {
-		return []error{errors.Wrap(err, "failed to marshal content")}
+		return []AdmissionError{{Error: errors.Wrap(err, "failed to marshal content")}}
 	}
+	admissionErrors := []AdmissionError{}
 	_, errs := rulefmt.Parse(content)
-	return errs
+
+	for _, err := range errs {
+		switch promErr := err.(type) {
+		case *rulefmt.Error:
```

Comment:
Indeed I forgot that part, maybe it's best to add a default
```golang
		default:
			admissionError := AdmissionError{
				Field: groupTag,
				Error: err,
			}
			admissionErrors = append(admissionErrors, admissionError)
		}
```
Since if it's not a `rulefmt.Error` we will not be able to tell anything else from the field apart from that it happened somewhere at the`groups` level.